src/Program.ts

```diff
@@ -705,42 +779,73 @@ export class Program {
+                            //remove last .
+                            if (dotPart.endsWith('.')) {
+                                dotPart = dotPart.substring(0, dotPart.length - 1);
+                            }
+                        }
+                        let statements = file.program.getStatementsByName(name, file, dotPart);
+                        for (let statement of statements) {
+                            //TODO better handling of collisions - if it's a namespace, then don't show any other overrides
+                            //if we're on m - then limit scope to the current class, if present
+                            let sigHelp = statement.file.getSignatureHelp(statement.item);
+                            if (sigHelp && !results.has[sigHelp.key]) {
+                                sigHelp.index = expr.args.findIndex((e) => util.rangeContains(e.range, position));
+                                results.set(sigHelp.key, sigHelp);
+                            }
+                        }
                     }
-                };
-                file.parser.ast.walk(createVisitor({
-                    FunctionStatement: statementHandler,
-                    ClassMethodStatement: statementHandler
-                }), {
-                    walkMode: WalkMode.visitStatements
-                });
+                } else if (isCallfuncExpression(expr)) {
+                    for (const scope of this.getScopes()) {
+                        //to only get functions defined in interface methods
+                        const lowerName = expr.methodName.text.toLowerCase();
+                        const callable = scope.getAllCallables().find((c) => c.callable.name.toLowerCase() === lowerName);
+                        if (callable) {
+                            let sigHelp = (callable.callable.file as BrsFile).getSignatureHelp(callable.callable.functionStatement);
+                            if (sigHelp && !results.has[sigHelp.key]) {
+                                sigHelp.index = expr.args.findIndex((e) => util.rangeContains(e.range, position));
+                                results.set(sigHelp.key, sigHelp);
+                            }
+                        }
+                    }
+                } else if (isNewExpression(expr)) {
+                    const nameParts = expr.className.getNameParts();
+                    let clazzItem = file.getClassByName(nameParts[nameParts.length - 1], nameParts.slice(0, -1).join('.'));
+                    let sigHelp = clazzItem?.file?.getClassSignatureHelp(clazzItem?.item);
+                    if (sigHelp && !results.has(sigHelp.key)) {
+                        sigHelp.index = expr.call.args.findIndex((e) => util.rangeContains(e.range, position));
+                        results.set(sigHelp.key, sigHelp);
+                    }
+                }
             }
-        }
+        };
 
-        return results;
+        file.parser.ast.walk(createVisitor({
```

Comment:
Honestly, I don't think you need to do a visitor here. You can loop over `parser.references` for these three things. that'll be WAY more efficient.

Reply:
was following the patterns I saw. I agree - now that I have better knoweldge of the references, that this is a faster choice.. I can fix that later this week/weekend.

Reply:
So `.references` only has `NewExpression`s. We don't have ` we don't actually have function calls or callfuncs. Meaning this is currently the best approach for finding these, unless we update the parser to capture more things in `references.

Reply:
I'd like to warn that walking all the expressions of a file like that is very costly - you don't even use the cancellation pattern to stop walking when you have found a match. It might be very bad for people with large files - they already suffer quite a bit just with the effort it takes for completion to parse the file's AST.

You should try to narrow what you walk: for brs code, just find the overall function the user is in, and for a class, find the class member the position is in - they you walk this element only.